### ISSUE TITLE
Boot2Docker 1.5 is not properly forwarding ports

### ISSUE THREAD

**Issue by jwaldrip** [boot2docker/boot2docker#788]
I am trying to run [this app](https://github.com/brandfolder/flynn-test-app). I am not able to hit it from my mac using the docker host ip and port or via curl when using `boot2docker ssh`.

### Build (or use docker-compose build)
`docker build -t flynn-test .`

### Run (or use docker-compose up)
`docker run -t -p 9292:9292 flynn-test .`

### Try to curl locally
```sh
$ curl http://192.168.59.103:9292/
$ curl: (7) Failed to connect to 192.168.59.103 port 9292: Connection refused
```

### Try to curl with boot2docker ssh
```sh
$ boot2docker ssh
docker@boot2docker:~$ curl http://127.0.0.1:9292
curl: (7) Failed connect to 127.0.0.1:9292; Connection refused
```

**Comment by upio** [comment]
I'm also getting this problem. Everything is set up fine like you've described, but it just refuses to work.

**Comment by detierno** [comment]
Im having the same problem, but i really new to docker and boot2docker to help with further information

**Comment by upio** [comment]
So I was having this problem but it was nothing to do with Docker. The reason it wasn't working or me is because my app was listening on "localhost" instead of 0.0.0.0. Make sure your app is doing that.

I had a quick look at that the Ruby app @jwaldrip linked to. It is listening on port 8080. Why are you trying to reach it on port 9292?

config.json

    "web": {
        "cmd": ["rackup", "-p$PORT", "-h$HOST"],
        "ports": [{
            "port": 8080,
            "proto": "tcp"
        }]
    }

Dockerfile

    ENV PORT 8080
    ENV HOST 0.0.0.0
    EXPOSE 8080

Have you tried running:

    docker run -t -p 8080:8080 flynn-test .
    boot2docker ssh
    curl http://localhost:8080/

**Comment by SvenDowideit** [comment]
yup, this is a common problem with services defaulting to listening to the localhost only :(
